File: Source/Policies/Algorithms.cs
using HmacManager.Components;

namespace HmacManager.Policies;

public class Algorithms
{
    public ContentHashAlgorithm ContentHashAlgorithm { get; set; } = ContentHashAlgorithm.SHA256;
    public SigningHashAlgorithm SigningHashAlgorithm { get; set; } = SigningHashAlgorithm.HMACSHA256;
}

File: src/Scripts/HmacManager/policies/algorithms.ts
export type ContentHashAlgorithm = "SHA1" | "SHA256" | "SHA512";
export type SigningHashAlgorithm = "HMACSHA1" | "HMACSHA256" | "HMACSHA512";
export class Algorithms {
    contentHashAlgorithm: ContentHashAlgorithm;
    signingHashAlgorithm: SigningHashAlgorithm;
}